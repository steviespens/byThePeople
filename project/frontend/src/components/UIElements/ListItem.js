import React, { Component } from "react";


function ListItem(props) {

    return (
        <div className='list-item' role="button" onClick={props.onClick}>
            <div className='list-item-inner-div'>
                {props.children}
                <span className='list-item-inner-span'>
                    
                    <p className='p1'>{props.primary}</p>
                    <p className='p2'>{props.secondary}</p>
                </span>
            </div>
        </div>

    );

}
export default ListItem;

