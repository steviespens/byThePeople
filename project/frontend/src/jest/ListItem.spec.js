import ListItem from "../components/UIElements/ListItem";
import React from "react";
import { shallow } from 'enzyme';

describe('<ListItem/>', () => {
    it('Renders without crashing', () => {
        let component = shallow(<ListItem></ListItem>);
        expect(component.type()).toEqual('div');
        expect(component).toMatchSnapshot();
    });
    it('check props primary and secondary', () => {
        let props = {
            primary: 'Primary',
            secondary: 'Secondary'
        },
        component = shallow(<ListItem {...props}></ListItem>);
        expect(component.find('p').length).toEqual(2);
        expect(component.find('p').at(0).text()).toEqual('Primary');
        expect(component.find('p').at(1).text()).toEqual('Secondary');
    });
    it('check onClick fires', () => {
        let mockFn = jest.fn(),
        component = shallow(<ListItem onClick={mockFn}></ListItem>);
        component.simulate('click');
        expect(mockFn).toHaveBeenCalled();
    })
});
