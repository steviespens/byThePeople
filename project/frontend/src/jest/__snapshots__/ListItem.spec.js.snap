// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem/> Renders without crashing 1`] = `ShallowWrapper {}`;
